wallet/src/Dscp/Wallet/Knit.hs

```diff
@@ -0,0 +1,170 @@
+module Dscp.Wallet.Knit where
+
+import Prelude hiding (preview)
+
+import Codec.Serialise (deserialiseOrFail)
+import Control.Exception
+import Control.Lens
+import Dscp.Core (addrFromText, txId)
+import Dscp.Crypto (mkPassPhrase)
+import IiExtras
+import Text.Earley
+
+import qualified Data.ByteString.Lazy as BSL
+import qualified Serokell.Util.Base64 as Base64
+import qualified Text.Megaparsec as P
+import qualified Text.Megaparsec.Char as P
+import qualified Text.Megaparsec.Char.Lexer as P
+
+import Dscp.Wallet.Face
+
+import Knit
+
+-- Component type for Knit
+data Wallet
+
+data instance ComponentValue _ Wallet
+  = ValueAddress Address
+  | ValueCoin Coin
+  | ValueSecretKey ByteString
+  deriving (Eq, Ord, Show)
+
+makePrisms 'ValueAddress
+
+instance Elem components Wallet => ComponentInflate components Wallet where
+  componentInflate = \case
+    ValueAddress a -> ExprLit $ toLit (LitAddress a)
+    ValueCoin c -> ExprLit $ toLit (LitCoin c)
+    ValueSecretKey sk -> ExprLit $ toLit (LitSecretKey sk)
+
+data instance ComponentLit Wallet
+  = LitAddress Address
+  | LitCoin Coin
+  | LitSecretKey ByteString
+  deriving (Eq, Ord, Show)
+
+data instance ComponentToken Wallet
+  = TokenAddress Address
+  | TokenCoin Coin
+  | TokenSecretKey ByteString
+  deriving (Eq, Ord, Show)
+
+makePrisms 'TokenAddress
+
+instance Elem components Wallet => ComponentTokenizer components Wallet where
+  componentTokenizer =
+      [ toToken . TokenAddress <$> pAddress
+      , toToken . TokenCoin <$> pCoin
+      , toToken . TokenSecretKey <$> pSecretKey
+      ]
+    where
+      pAddress :: Tokenizer Address
+      pAddress = do
+        str <- pSomeAlphaNum
+        either (fail . show) return $ addrFromText str
+
+      pCoin :: Tokenizer Coin
+      pCoin = Coin <$> P.signed (return ()) P.decimal
```

Comment:
`Coin` is still unsigned

Reply:
Right, I forgot it's `Word64`, not `Scientific`